torchtune/utils/_distributed.py

```diff
@@ -242,6 +250,79 @@ def lora_wrap_fsdp(module: nn.Module, recurse: bool, **kwargs):
     return lora_wrap_fsdp
 
 
+def load_from_full_model_state_dict(
+    model: FSDPModule,
+    full_sd: Dict[str, Any],
+    device: torch.device,
+    is_rank_zero: bool,
+):
+    meta_sharded_sd = model.state_dict()
+    sharded_sd = {}
+    for param_name, full_tensor in full_sd.items():
+        sharded_meta_param = meta_sharded_sd.get(param_name)
+        sharded_tensor = distribute_tensor(
+            full_tensor, sharded_meta_param.device_mesh, sharded_meta_param.placements
```

Comment:
Where is device_mesh and placements information coming from?

Reply:
it's from `fully_shard(model, mesh)`. we are using default `mesh` where every rank serves FSDP, since there is no 2D/3D parallasim involved

after `fully_shard(model, mesh)`, `model.parameters()` is converted from plain tensor to `DTensor` with `mesh`